javaee/impl/src/main/java/org/jboss/forge/addon/javaee/jpa/ui/JPANewEntityCommandImpl.java

```diff
@@ -86,13 +108,31 @@ public JavaClassSource decorateSource(UIExecutionContext context, Project projec
       {
          idStrategyChosen = GenerationType.AUTO;
       }
+
+      EntityIdType idTypeChosen = idType.getValue();
+      if(idTypeChosen == null){
+          idTypeChosen = EntityIdType.LONG_PROPERTY;
+      }
+
       ConfigurationFacet facet = project.getFacet(ConfigurationFacet.class);
       Configuration config = facet.getConfiguration();
       String idPropertyName = config.getString(PersistenceOperations.ID_PROPERTY_NAME_CONFIGURATION_KEY,
                configuration.getString(PersistenceOperations.ID_PROPERTY_NAME_CONFIGURATION_KEY, "id"));
       String versionPropertyName = config.getString(PersistenceOperations.VERSION_PROPERTY_NAME_CONFIGURATION_KEY,
                configuration.getString(PersistenceOperations.VERSION_PROPERTY_NAME_CONFIGURATION_KEY, "version"));
-      return persistenceOperations.newEntity(source, idStrategyChosen, tableName.getValue(), idPropertyName,
-               versionPropertyName);
+
+      switch(idTypeChosen){
+          case LONG_PROPERTY:
+              return persistenceOperations.newEntity(source, idStrategyChosen, tableName.getValue(), idPropertyName,
+                      versionPropertyName);
+          case EMBEDDED_ID:
+              // XXX what if idClass value is empty?
+              return persistenceOperations.newEntityEmbeddedId(source, tableName.getValue(), idPropertyName,
+                      idClass.getValue(), versionPropertyName);
+          case ID_CLASS:
```

Comment:
In this case perhaps we should remove from the enum?

Reply:
`idClass` and `ID_CLASS` are unrelated.
`idClass` is the Class/Type to use for the `@EmbeddedId`, `ID_CLASS` means to use the `@IdClass`-Annotation (at the Entity), in contrast to `EMBEDDED_ID` which means `@EmbeddedId` (at the property).

Or Do you mean the `UnsupportedOperationException`?
That one is only temporary until I worked out how I want to do `@IdClass`.

Reply:
Ah right, I thought that wasn't temporary, that's fine :+1: